### ISSUE TITLE
Can't merge pull requests when commit signing is required.

### ISSUE THREAD

**Issue by WildCryptoFox** [go-gitea/gitea#12272]
(Using gitea.com)

## Description

I've created a PR under a repository which requires commit signing but Gitea refuses to allow me to merge because the merge commit itself won't be signed and there is no associated key for signing the merge commit.

GitLab allows for [fast-forward merging without explicit merge commits](https://docs.gitlab.com/ee/user/project/merge_requests/fast_forward_merge.html); this is just `--ff-only`. Please add this as an option. Not everyone wants explicit merge commits.

It would also help if Gitea presented instructions for manually fetching and applying merges from the command line without logging into Gitea and without requiring merge commits.

Thank you.

## Screenshots

![gitea](https://user-images.githubusercontent.com/637155/87865072-fe13ad80-c95f-11ea-8711-404cac90a4df.png)

**Comment by lunny** [comment]
This is because gitea.com didn't config the signing key.

**Comment by WildCryptoFox** [comment]
@lunny Yes, but I and others still want `--ff-only` merges without merge commits.

**Comment by WildCryptoFox** [comment]
To make this worse; with protected branches it can't be merged manually either.

**Comment by WildCryptoFox** [comment]
@lunny I've merged my test PR by disabling the signed commits constraint and using "rebase merge". Git accepted it could do the fast-forward and didn't add a merge commit. The `--ff-only` option only requires the PR be fast-forwardable such that it doesn't need to add the merge commit and doesn't conflict with the forced commit signing.

**Comment by WildCryptoFox** [comment]
The issue remains.

**Comment by zeripath** [comment]
Can you replicate on try?

**Comment by WildCryptoFox** [comment]
@zeripath [Reproduced](https://try.gitea.io/12272/test/pulls/1)

**Comment by WildCryptoFox** [comment]
Note: This option under branch protections implies `--ff-only` and we do not want Gitea to sign any merge commits. We only want `-ff-only` merges.

> Block merge if pull request is outdated.
> Merging will not be possible when head branch is behind base branch

**Comment by zeripath** [comment]
could you make arandomer an admin collaborator so I can check the repository settings please.

**Comment by zeripath** [comment]
If you have set this:

![Screenshot from 2020-09-28 09-08-59](https://user-images.githubusercontent.com/1824502/94406655-41fa0e80-016a-11eb-9dbe-b830093e9537.png)

That would be a reason why you can't merge using Gitea when there is no signing key available. 

The constraint for signed commits applies absolutely.

**Comment by lafriks** [comment]
@zeripath but that should still allow `--ff-only` merges as they do not require Gitea to sign anything

**Comment by WildCryptoFox** [comment]
@zeripath Requiring signed commits is exactly what I want and as @lafriks concurs, should not interfere with `--ff-only` because there is no merge commit to sign.

**Comment by jvisser-di** [comment]
I've run into this. What are the options for me to work around this?

**Comment by MrMetacomCode** [comment]
I have this issue as well. 

I tried verifying my GPG key on my account to see if it would fix it, but it said the signature I was using to verify the token wasn't correct, even though I verified the signature was correct with GPG before I gave it back to Gitea.

I've just turned off requiring signed commits so I can continue development at this point.

**Comment by krjakbrjak** [comment]
Take a look at #37335
